CDPSDX-3991: Removed DL flow for submitting datalake data sizes.

sxxgrc opened the pull request:
Jira: https://jira.cloudera.com/browse/CDPSDX-3991

Simply removed the original flow and just route the endpoint method to directly call the DR endpoint submission method. This is a very lightweight process and we do not wait on a response from the DR service so this should not have any undesirable effects.

Main reason to do this is that we have observed that this clashes with flows that call backup such as the upgrade validation flow.

I also added a quick conditional check to the DB backup flow chain creation so as to avoid re-running the salt update after it was already updated for determine DL data sizes.

votez commented on core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/chain/BackupDatalakeDatabaseFlowEventChainFactory.java:
```diff
@@ -32,8 +40,11 @@ public String initEvent() {
-        flowEventChain.add(new StackEvent(SaltUpdateEvent.SALT_UPDATE_EVENT.event(), event.getResourceId(), event.accepted()));
-        flowEventChain.add(new DatabaseBackupTriggerEvent(DATABASE_BACKUP_EVENT.event(), event.getResourceId(),
+        StackStatus stackStatus = stackService.getCurrentStatusByStackId(event.getResourceId());
+        if (!stackStatus.getDetailedStackStatus().equals(DETERMINE_DATALAKE_DATA_SIZES_FINISHED)) {
```
is this the optimization not to invoke salt update too often?

sxxgrc replied:
Yes it is, with this the DB backup will not invoke the salt update if the determine datalake data sizes flow was already run.

sxxgrc commented on core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/datalake/metrics/datasizes/DetermineDatalakeDataSizesActions.java:
```diff
@@ -65,6 +69,7 @@ protected Selectable createRequest(DetermineDatalakeDataSizesContext context) {
+                LOGGER.info("Submitting datalake data sizes info '{}' for cluster {}.", payload.getDataSizesResult(), stack.getResourceCrn());
                 String initiatorUserCrn = ThreadBasedUserCrnProvider.getUserCrn();
                 ThreadBasedUserCrnProvider.doAsInternalActor(
                         regionAwareInternalCrnGeneratorFactory.iam().getInternalCrnForServiceAsString(),
```
I do not believe we have any explicit quidelines for this. I believe we usually handle things in a handler when we need to perform polling on an external call, but I think just making an external call with no waiting is fine here.

votez replied:
we have [guidelines](https://github.com/hortonworks/cloudbreak/tree/master/flow) under the "General hints" section. But we can estimate this as is not too long operation so acceptable for the Action.

lacikaaa commented on core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/chain/BackupDatalakeDatabaseFlowEventChainFactory.java:
```diff
@@ -32,8 +40,11 @@ public String initEvent() {
-        flowEventChain.add(new StackEvent(SaltUpdateEvent.SALT_UPDATE_EVENT.event(), event.getResourceId(), event.accepted()));
-        flowEventChain.add(new DatabaseBackupTriggerEvent(DATABASE_BACKUP_EVENT.event(), event.getResourceId(),
+        StackStatus stackStatus = stackService.getCurrentStatusByStackId(event.getResourceId());
+        if (!stackStatus.getDetailedStackStatus().equals(DETERMINE_DATALAKE_DATA_SIZES_FINISHED)) {
```
maybe add a log around this so it would be easier to investigate what happens

lacikaaa reviewed (changes requested):
- please make sure the commit message contains all the relevant info (GH desc doesn't count)
- please ensure you have unit tests for your changes. I don't see any unit test changes

sxxgrc commented:
@lacikaaa I have addressed your comments, adding a new unit test, and updating the commit message to be the same as the description for the PR. Thanks.
